[Clang-10] Direct list initialization of std::variant fails in IRGeneratorForStatements.cpp

https://github.com/ethereum/solidity/blob/2d1c4b770fc884986450f3f12780644a06e9f7f2/libsolidity/codegen/ir/IRGeneratorForStatements.cpp#L969-L972

Clang 10 complains that 

```
/src/solidity/libsolidity/codegen/ir/IRGeneratorForStatements.cpp:971:5: error: no viable conversion from 'int' to 'const std::variant<std::string, unsigned int>' (aka 'const variant<basic_string<char, char_traits<char>, allocator<char> >, unsigned int>')
                                0
                                ^
/usr/local/include/c++/v1/variant:1195:3: note: candidate constructor not viable: no known conversion from 'int' to 'const std::__1::variant<std::__1::basic_string<char>, unsigned int> &' for 1st argument
   variant(const variant&) = default;
  ^
/usr/local/include/c++/v1/variant:1196:3: candidate constructor not viable: no known conversion from 'int' to 'std::__1::variant<std::__1::basic_string<char>, unsigned int> &&' for 1st argument
   variant(variant&&) = default;
  ^
/usr/local/include/c++/v1/variant:1208:13: candidate template ignored: substitution failure [with _Arg = int, $1 = 0, $2 = 0, $3 = 0]: no type named 'type' in 'std::__1::invoke_result<std::__1::__variant_detail::__all_overloads<std::__1::__variant_detail::__overload<std::__1::basic_string<char>, 0>, std::__1::__variant_detail::__overload<unsigned int, 1> >, int, int>'
   constexpr variant(_Arg&& __arg) noexcept(
```

--- a/libsolidity/codegen/ir/IRGeneratorForStatements.cpp
+++ b/libsolidity/codegen/ir/IRGeneratorForStatements.cpp
@@ -968,7 +968,7 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 			*_indexAccess.annotation().type,
 			IRLValue::Storage{
 				slot,
-				0
+				0u
 			}
 		});
 	}

